src/utilities.cpp

```diff
@@ -463,4 +463,99 @@ size_t getVariableIndexInComponent(const ComponentPtr &component, const Variable
     return index;
 }
 
+bool isEntityChildOf(const EntityPtr &entity1, const EntityPtr &entity2)
+{
+    return entity1->parent() == entity2;
+}
+
+bool areEntitiesSiblings(const EntityPtr &entity1, const EntityPtr &entity2)
+{
+    auto entity1Parent = entity1->parent();
+    return entity1Parent != nullptr && entity1Parent == entity2->parent();
+}
+
+using PublicPrivateRequiredPair = std::pair<bool, bool>;
+
+/**
+ * @brief Determine whether a public and/or private interface is required for the givn @p variable.
+ *
+ * Determine whether a public and/or private interface is required for the givn @p variable.  Returns
+ * a pair of booleans where the first item in the pair indicates that a public interface is required,
+ * the second item in the pair indicates that a private interface is required, and if both are items
+ * in the pair are false then this indicates an error has occured and the interface type cannot be
+ * determined.
+ *
+ * @param variable The variable to detect the interface type required.
+ *
+ * @return A pair of booleans.
+ */
+PublicPrivateRequiredPair publicAndOrPrivateInterfaceTypeRequired(const VariablePtr &variable)
+{
+    PublicPrivateRequiredPair pair = std::make_pair(false, false);
+    for (size_t index = 0; index < variable->equivalentVariableCount() && !(pair.first && pair.second); ++index) {
+        auto equivalentVariable = variable->equivalentVariable(index);
+        auto componentOfVariable = variable->parent();
+        auto componentOfEquivalentVariable = equivalentVariable->parent();
+        if (componentOfVariable == nullptr || componentOfEquivalentVariable == nullptr) {
+            return std::make_pair(false, false);
+        }
+        if (areEntitiesSiblings(componentOfVariable, componentOfEquivalentVariable)) {
+            pair.first = true;
+        } else if (isEntityChildOf(componentOfVariable, componentOfEquivalentVariable)) {
+            pair.first = true;
```

Comment:
Merge these two tests since they both set `pair.first` to `true`...?

Reply:
I considered doing this but thought it was clearer and cleaner to separate it out.

Reply:
Not sure it is, but it's not critical, so I won't insist. :)